### PR TITLE
Raft test topology II part 1

### PR DESCRIPTION
**Pull request by alecco**
- Remove `ScyllaCluster.__getitem__()`  (pending request by @kbr- in a previous pull request), for this remove all direct access to servers from caller code
- Increase Python driver timeouts (req by @nyh)
- Improve `ManagerClient` API requests: use `http+unix://<sockname>/<resource>` instead of `http://localhost/<resource>` and callers of the helper method only pass the resource
- Improve lint and type hints

### CODE REVIEW

**Review by nyh — APPROVED** [review]

**Comment by nyh** [comment]
Looks good to me. I'm waiting for CI to run on this before merging -  I don't know why it didn't even start yet.

**Comment by nyh** [comment]
@benipeled the CI never started on this PR - which was created three hours ago. Any idea why? Thanks.

**Comment by alecco** [comment]
> @benipeled the CI never started on this PR - which was created three hours ago. Any idea why? Thanks.

@nyh I see a job:

https://jenkins.scylladb.com/job/releng/job/Scylla-CI/1825/

**Comment by nyh** [comment]
> > @benipeled the CI never started on this PR - which was created three hours ago. Any idea why? Thanks.
> 
> @nyh I see a job:
> 
> https://jenkins.scylladb.com/job/releng/job/Scylla-CI/1825/

Good. I guess it's just that the bot forgot somehow to add the link in the PR?

**Comment by nyh** [comment]
According to https://jenkins.scylladb.com/job/releng/job/Scylla-CI/1825/console, this job never got to run, it has been waiting for the last three hours for  a chance to run. Why? Anyway, maybe this is why we're not seeing anything here - maybe the job itself, when it finally starts to run, is the one responsible to update the PR.

**Comment by alecco** [comment]
> > > @benipeled the CI never started on this PR - which was created three hours ago. Any idea why? Thanks.
> > 
> > 
> > @nyh I see a job:
> > https://jenkins.scylladb.com/job/releng/job/Scylla-CI/1825/
> 
> Good. I guess it's just that the bot forgot somehow to add the link in the PR?

I think it only posts on success. I usually go check at the Jenkins console https://jenkins.scylladb.com/job/releng/job/Scylla-CI/

But when it's building it doesn't show the PR number so you have to click on the builds. You can find by date/hour  ¯\_(ツ)_/¯

**Comment by kbr-** [comment]
The PR state gets updated by the job at some point after it starts running, you're able to see the status at the bottom of  the PR page.

**Comment by benipeled** [comment]
There was an issue with the infra of external-resource used by the cloud-init use for the creation of AWS instances (yes, it's a mess), It was fixed and everything should work as expected.

**Pull request merged**
